File: 7KYU/Get the MIddle Character/solution.go
package kata

import (
	"fmt"
)

func GetMiddle(s string) string {
	if len(s)%2 == 0 {
		half := len(s) / 2
		return fmt.Sprint(s[half-1 : half+1])
	} else {
		half := int(len(s) / 2)
		return string(s[half])
	}
}

File: 7KYU/Get the MIddle Character/solution.cpp
#include <string.h>
using namespace std;

string get_middle(string input) 
{
    if(input.length() % 2 == 0) {
      auto half = input.length() / 2;
      return input.substr(half-1, 2);
    } else {
      int half = (input.length() / 2);
      return input.substr(half, 1);
    }
}